Documentation/network/clustermesh/mcsapi.rst

```diff
@@ -0,0 +1,165 @@
+.. _gs_clustermesh_mcsapi:
+
+*********************************
+Multi-Cluster Services API (Beta)
+*********************************
+
+.. include:: ../../beta.rst
+
+This tutorial will guide you to through the support of `Multi-Cluster Services API (MCS-API) <MCS_API_KEP>`_ in Cilium.
+
+.. _MCS_API_KEP: https://github.com/kubernetes/enhancements/blob/master/keps/sig-multicluster/1645-multi-cluster-services-api/README.md
+
+Prerequisites
+#############
+
+You need to have a functioning Cluster Mesh setup, please follow the guide
+:ref:`gs_clustermesh` to set it up.
+
+You first need to install the required MCS-API CRDs:
+
+   .. code-block:: shell-session
+
+      kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/mcs-api/refs/heads/4690877cc89bcfd9302d516ae43515f446d0aecf/config/crd/multicluster.x-k8s.io_serviceexports.yaml
+      kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/mcs-api/4690877cc89bcfd9302d516ae43515f446d0aecf/config/crd/multicluster.x-k8s.io_serviceimports.yaml
+
+
+To install Cilium with MCS-API support, run:
+
+   .. code-block:: shell-session
+
+      helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set clustermesh.enableMCSAPISupport=true
+
+To enable MCS-API API support on an existing Cilium installation, run:
+
+   .. code-block:: shell-session
+
+      helm upgrade cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --reuse-values \\
+      --set clustermesh.enableMCSAPISupport=true
+      kubectl -n kube-system rollout restart deployment/cilium-operator
+      kubectl -n kube-system rollout restart deployment/clustermesh-apiserver
+
+Also checkout the `EndpointSlice synchronization <endointslicesync>`_ feature
+if you need Headless Services support.
+
+Installing CoreDNS multicluster
+-------------------------------
+
+You also need to install and configure the `multicluster CoreDNS plugin <multicluster_coredns>`_.
+
+.. _multicluster_coredns: https://github.com/kubernetes/enhancements/blob/master/keps/sig-multicluster/1645-multi-cluster-services-api/README.md
+
+The multicluster CoreDNS plugin is currently not provided by default in
+CoreDNS so you will have to recompile it yourself.
```

Comment:
I'd maybe add a step to clone the CoreDNS repo, just to make super clear the context of the subsequent operations.